snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -683,58 +662,70 @@ func (b *Bootstrapper) process(ctx context.Context, blk snowman.Block, processin
 
 		// If the block wasn't able to be acquired immediately, attempt to fetch
 		// it
-		b.Blocked.AddMissingID(parentID)
 		if err := b.fetch(ctx, parentID); err != nil {
 			return err
 		}
 
-		if err := b.Blocked.Commit(); err != nil {
-			return err
-		}
-		return b.tryStartExecuting(ctx)
```

Comment:
Calling `tryStartExecuting` multiple times in a call to the engine can cause the engine to think that it has finished fetching blocks multiple times. This means that the engine may finish multiple times - which means it may exit early than it should (may exit after only one round of bootstrapping or finish even though we were still catching up to tip).